channels/generic/http.py

```diff
@@ -43,6 +43,9 @@ def __init__(self, *args, **kwargs):
         await self.send(
             {"type": "http.response.body", "body": body, "more_body": more_body}
         )
+        if not more_body:
+            await self.disconnect()
+            raise StopConsumer()
```

Comment:
This should be Daphne's job. (Via `http_disconnect`)

Reply:
@carltongibson in this case, for checking if the body is fully sent/empty, is the expectation/correctness that `await self.disconnect()` is the right move? Or is the disconnect & StopConsumer the wrong flow?

I'm curious for how we signal up to the event loop manager (e.g. daphne, uvicorn) that we're ready for the disconnect to take plase. I can see why he considered doing the `raise` but unsure what the solution should be.

Reply:
Hi @jheld. 

> I'm curious for how we signal up to the event loop manager (e.g. daphne, uvicorn) that we're ready for the disconnect to take plase. 

My thought was that Daphne already knows that the response has finished at this point:

https://github.com/django/daphne/blob/59b57a9f4b7a419f1ab3ecb792ca802671045075/daphne/http_protocol.py#L256-L258

I think :) -- it was a couple of months ago when I looked at this.

Reply:
@jheld See the discussion on #1255: I think that´s more telling that.